detox/e2e/test/notifications/mention_badges.e2e.js

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// *******************************************************************
+// - [#] indicates a test step (e.g. # Go to a screen)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element testID when selecting an element. Create one if none.
+// *******************************************************************
+
+import {MainSidebar} from '@support/ui/component';
+import {ChannelScreen} from '@support/ui/screen';
+import {
+    Channel,
+    Post,
+    Team,
+    User,
+} from '@support/server_api';
+
+describe('Mention Badges', () => {
+    const {
+        channelNavBarTitle,
+        closeMainSidebar,
+        goToChannel,
+        mainSidebarDrawerButtonBadgeUnreadCount,
+        openTeamSidebar,
+    } = ChannelScreen;
+    const {
+        closeTeamSidebar,
+        getTeamByDisplayName,
+        switchTeamsButtonBadgeUnreadCount,
+    } = MainSidebar;
+    let testChannel;
+    let testPrivateChannel;
+    let testPublicChannel;
+    let testTeam1;
+    let testTeam2;
+    let testUser1;
+    let testUser2;
+
+    beforeAll(async () => {
+        ({user: testUser1} = await User.apiCreateUser());
+        ({team: testTeam1} = await Team.apiCreateTeam({prefix: 'team-a'}));
+        ({channel: testChannel} = await Channel.apiGetChannelByName(testTeam1.id, 'town-square'));
+
+        ({user: testUser2} = await User.apiCreateUser());
+        ({team: testTeam2} = await Team.apiCreateTeam({prefix: 'team-b'}));
+        ({channel: testPublicChannel} = await Channel.apiGetChannelByName(testTeam2.id, 'town-square'));
+
+        await Team.apiAddUserToTeam(testUser1.id, testTeam1.id);
+        await Team.apiAddUserToTeam(testUser1.id, testTeam2.id);
+        await Team.apiAddUserToTeam(testUser2.id, testTeam1.id);
+        await Team.apiAddUserToTeam(testUser2.id, testTeam2.id);
+
+        ({channel: testPrivateChannel} = await Channel.apiCreateChannel({type: 'P', teamId: testTeam2.id}));
+
+        await Channel.apiAddUserToChannel(testUser1.id, testPrivateChannel.id);
+        await Channel.apiAddUserToChannel(testUser2.id, testPrivateChannel.id);
+
+        // # Open channel screen
+        await ChannelScreen.open(testUser1);
+
+        // # Clear unreads and go back to original team and channel
+        await openTeamSidebar();
+        await getTeamByDisplayName(testTeam2.display_name).tap();
+        await goToChannel(testPrivateChannel.display_name);
+        await openTeamSidebar();
+        await getTeamByDisplayName(testTeam1.display_name).tap();
+    });
+
+    afterAll(async () => {
+        await ChannelScreen.logout();
+    });
+
+    const expectMentionBadges = async (messagePostedChannel) => {
```

Comment:
nit: Would be nice to see the two `it` blocks first and move this function below.

Reply:
Updated.